libethereum/Executive.cpp

```diff
@@ -199,10 +199,14 @@ bool Executive::go(OnOpFunc const& _onOp)
 
 			if (m_isCreation)
 			{
-				if (m_out.size() * c_createDataGas <= m_endGas)
-					m_endGas -= m_out.size() * c_createDataGas;
+				bigint storageCost = (bigint)m_out.size() * c_createDataGas;
```

Comment:
bigint is unbound. m_out.size() is of type size_t , which is long unsigned int. As long as c_createDataGas stays at this order of magnitude (=200 right now) then you can not overflow by doing the multiplication (200 * size_t.max can never be larger than u256.max), but the you can overflow the size() function by returning m_out with a real size larger than size_t.max. I will write a test for that.